controllers/tokenupload_controller.go

```diff
@@ -0,0 +1,229 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"time"
+
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	"github.com/go-logr/logr"
+
+	"github.com/redhat-appstudio/service-provider-integration-operator/pkg/spi-shared/tokenstorage"
+	"k8s.io/apimachinery/pkg/types"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	spi "github.com/redhat-appstudio/service-provider-integration-operator/api/v1beta1"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const tokenSeretLabel = "spi.appstudio.redhat.com/token"
```

Comment:
Well, it can be "spi.appstudio.redhat.com/secret-type"="token"  , I'd say it is pretty much the same.
On one hand, I agree it looks nicer to have not empty label value, on the other hand I am not sure we use a "secret" word in our project vocabulary, unlike "token" word. So It may be some confusing from terminology standpoint...
WDYT?

As for moving spiTokenName (and providerURL too) to the label, we can do that since it probably fits the labels intention: "to be used to specify identifying attributes of objects that are meaningful and relevant to users, but do not directly imply semantics to the core system." 
+1